### ISSUE TITLE
W3C XML Conformance Test Suite

### ISSUE THREAD

**Issue by rushmorem** [RazrFalcon/xmlparser#1]
I noticed that the parser is parsing some invalid XML as valid. For example, incorrect nesting:

```xml
<root><child></root></child>
```

If it's not too much trouble, I think it would be great if it can be tested against the [W3C conformance test suite](https://www.w3.org/XML/Test/).

**Comment by RazrFalcon** [comment]
Yes. The parser doesn't validate the tree structure. It's up to you.

>I think it would be great if it can be tested against the W3C conformance test suite.

The library has about 150 tests already. Including some W3C tests.

**Comment by rushmorem** [comment]
I see. Thanks for the explanation. Do you plan to have validation as part of this library, or it's something that will have to be done by a higher level one?

**Comment by RazrFalcon** [comment]
I see. Also, unlike the `xml-rs`, this library is very fast. Usually ~10% slower that `quick-xml`.
